src/FlatRegistry.php

```diff
@@ -0,0 +1,148 @@
+<?php
+
+/**
+ * Part of the Joomla Framework Registry Package
+ *
+ * @copyright  Copyright (C) 2023 Open Source Matters, Inc.
+ * @license    GNU General Public License version 2 or later; see LICENSE
+ */
+
+namespace Joomla\Registry;
+
+/**
+ * Flat Registry class
+ *
+ * @since  2.1.0
+ */
+class FlatRegistry implements RegistryInterface
+{
+    /**
+     * Data storage
+     * @var array
+     */
+    private $data = [];
+
+    /**
+     * Check if a registry key exists.
+     *
+     * @param  string  $key  Registry key
+     *
+     * @return  bool
+     *
+     * @since   2.1.0
+     */
+    public function exists(string $key): bool
+    {
+        return array_key_exists($key, $this->data);
+    }
+
+    /**
+     * Get a registry value.
+     *
+     * @param  string  $key      Registry key
+     * @param  mixed   $default  Optional default value, returned if the internal value is null or empty string.
+     *
+     * @return  mixed  Value of entry or default value
+     *
+     * @since   2.1.0
+     */
+    public function get(string $key, $default = null)
+    {
+        $value = $this->data[$key] ?? null;
+
+        return ($value !== null && $value !== '') ? $value : $default;
+    }
+
+    /**
+     * Set a registry value.
+     *
+     * @param  string  $key    Registry key
+     * @param  mixed   $value  Value of entry
+     *
+     * @return  mixed  Previous value for the key.
+     *
+     * @since   2.1.0
+     */
+    public function set(string $key, $value)
+    {
+        $prevValue = $this->data[$key] ?? null;
+
+        $this->data[$key] = $value;
+
+        return $prevValue;
+    }
+
+    /**
+     * Delete a registry value.
+     *
+     * @param  string  $key  Registry key
+     *
+     * @return  mixed  The value of the removed node or null if not set
+     *
+     * @since   2.1.0
+     */
+    public function remove(string $key)
+    {
+        $prevValue = $this->data[$key] ?? null;
+
+        unset($this->data[$key]);
+
+        return $prevValue;
+    }
+
+    /**
+     * Return data as array.
+     *
+     * @return  array
+     *
+     * @since   2.1.0
+     */
+    public function toArray()
+    {
+        return $this->data;
+    }
+
+    /**
+     * Load data in to registry, will override an existing keys if exists.
+     *
+     * @param mixed $data  Iterable data to load, array or object
+     *
+     * @return static
+     *
+     * @since   2.1.0
+     */
+    public function loadData($data): RegistryInterface
```

Comment:
This appears to be the only function that may be causing unexpected issues. I am wondering if we can add more padding here, similar to the approach in https://github.com/joomla-framework/registry/blob/2.0-dev/src/Registry.php#L264 and https://github.com/joomla-framework/registry/blob/2.0-dev/src/Registry.php#L288, in order to set the data type with fewer surprises?

Reply:
Originaly I made 
```php
public function loadData(iterable $data)
```
But because it does not work with objects I have removed it.

`loadArray()` and `loadObject()` kind of the same. But can add it instead of `loadData()`.

Reply:
The presence of loadArray, loadObject, and loadFile enhances our control over the flow of data into the local array, ensuring that it is done in a consistent and predictable manner. By maintaining loadData and synthesizing the available options, we can direct the loading process to the most suitable function. This also provides developers with the option of accessing these functions directly if the data type is already known. Ultimately, the goal is to minimize the potential for unexpected edge cases. While there may not be a pressing need to retain loadData, having access to polymorphic functionality can be beneficial in certain scenarios.